pallets/template/src/lib.rs

```diff
@@ -231,34 +240,42 @@ pub mod pallet {
             Accounts::INVESTOR => {
                ensure!(InvestorLog::<T>::contains_key(&caller)==false,Error::<T>::OnlyOneRoleAllowed);
                ensure!(HouseSellerLog::<T>::contains_key(&caller)==false,Error::<T>::OnlyOneRoleAllowed);
```

Comment:
```suggestion
            Accounts::INVESTOR => {
               ensure!(InvestorLog::<T>::contains_key(&caller)==false,Error::<T>::OnlyOneRoleAllowed);
               ensure!(HouseSellerLog::<T>::contains_key(&caller)==false,Error::<T>::OnlyOneRoleAllowed);
```
In the selected code snippet above and the following roles below, it seems like this code can be simplified. I'm looking into the [alliance-pallet](https://github.com/paritytech/substrate/blob/bc310fdd1040af8a3d2901b5d6f4cf8d5250aeb7/frame/alliance/src/lib.rs#L411-L446). They are dealing also with roles and as roles get added you don't want to add different roles with `ensure` in there.

Reply:
Yeah, More roles will lead to more "ensure!()"...